sql-cli/sql_cli/__main__.py

```diff
@@ -218,28 +258,20 @@ def run(
     # TODO: Validate all DAG connections before running
 
     # Generate DAG before running
-    dag_file = _generate_dag(project=project, workflow_name=workflow_name, generate_tasks=generate_tasks)
+    tasks_to_execute = None
+    if task_id and not include_upstream:
+        tasks_to_execute = [task_id]
+    dag_file = _generate_dag(
```

Comment:
@pankajkoti, shouldn't we also pass `include_upstream` here to differentiate between running the single task - or also running its upstream dependencies?